go/libraries/doltcore/schema/alterschema/addcolumn.go

```diff
@@ -50,10 +50,6 @@ func AddColumnToTable(ctx context.Context, tbl *doltdb.Table, tag uint64, newCol
 		return nil, err
 	}
 
-	if err := validateNewColumn(ctx, tbl, tag, newColName, typeInfo, nullable, defaultVal); err != nil {
```

Comment:
Really should still do this validation here. In general, can't trust clients to do checks on their own.